Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# More History

## Changelog:
- Added a ton of new baronies.
- Added a lot of new counties.
- Added new duchies, to make sure every duchy has 5-3 counties. Not more.
- Added new kingdoms, to make sure every kingdom has 5-3 duchies. Not more.
- Made sure characters have 4-1 holdings, not over the limit.
- Added fords (over major rivers) and straits (over seas).


# How to test:
Check the history of all the dates. If you notice something weird or you want to change something, tell me.

# Notes:
- Don't force governments in the history files. It's bad! It makes you not able to use your own holdings. If you really want to change the government, change the holdings they own instead.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/115122250-f739ca00-9fc7-11eb-9e4c-3ffd412e7d16.png)

</details>

- Don't force laws in the history files. They were ported from CK2 where title laws played a bigger role. Now religions play this role thus we don't need them.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/115122333-6f07f480-9fc8-11eb-936c-c1e43971e2e8.png)

</details>

- I made the [Character Generator](https://github.com/Warcraft-GoA-Development-Team/CK3-Character-Generator/releases/tag/1.0.0), feel free to use it for generating more characters. Pay attention there are 2 tabs: **Characters** and **Titles**. **Characters** shows generated characters. **Titles** shows an inheritance line for their titles.

# Tasks
<details>
<summary>Click to expand</summary>

- [x] Clean forcing governments in history files. Read **Notes**.
- [x] Clean forcing laws in history files. Read **Notes**.
- [x] Rework the Barrens history. It needs a closer look. It's very different from what we had in CK2.
- [x] Write history for Un'goro
- [x] We should fix `landedtitlemanager.cpp:618`: county doesn't have a holder in history.
- [x] Lots of new history needed. Any province/dutchy marked with `#NEEDS NEW` requires new history. Some/many of them require new characters/holders as well.
  - [x] Eastern Kingdoms
  - [x] Northrend
  - [x] South Seas
  - [x] Kalimdor
  - [x] Pandaria
- [x] Since there are new dutchies/kingdoms, the liege will need to be updated in many places.
- [x] c_krasarang needs to be renamed
- [x] Counties that changes government type need to change holdings instead
- [x] Remove Drysnout duchy title
- [x] Restore the Defias Brotherhood
- [x] Restore the Dragonmaw
- [x] Generate Scourge rulers (a bit bunch of lands are controlled by Ner'zhul in 603).
- [x] Fix `landedtitlemanager.cpp:637`
- [x] `titlehistory.cpp:266`: Title holder dead
- [x] Keep some nerubs in Northrend
- [x] Make taurens independent from the Horde in 605
- [x] Populate Plunder Isle with pirates

</details>
